src/compiler/scala/tools/nsc/settings/Warnings.scala

```diff
@@ -24,6 +26,72 @@ trait Warnings {
   // Warning semantics.
   val fatalWarnings = BooleanSetting("-Werror", "Fail the compilation if there are any warnings.") withAbbreviation "-Xfatal-warnings"
 
+  private val WconfDefault = List("cat=deprecation:ws", "cat=feature:ws", "cat=optimizer:ws")
+  // Note: user-defined settings are added on the right, but the value is reversed before
+  // it's parsed, so that later defined settings take precedence.
+  val Wconf = MultiStringSetting(
+    "-Wconf",
+    "patterns",
+    "Configure reporting of compiler warnings; use `help` for details.",
+    helpText = Some(
+      s"""Configure compiler warnings.
+         |Syntax: -Wconf:<filter>&...&<filter>:<action>,<filter>:<action>,...
```

Comment:
```suggestion
         |Syntax: -Wconf:<filter>:<action>,<filter>:<action>,...
```

The `&` threw me off since I wasn't sure about the precedence of `&`, `:`, and `,`. Maybe `&` that can be added as a section below as "`<filter>` might be chained etc.".

Reply:
Thanks, that looks better. I'm also happy to change `&` for some other character, but this is the best I could come up with. 

The first `:` (in `-Wconf:`) and the `,` to split settings is given by the MultiStringSetting. There's also the "non-colonated-space-separated mode" that eats until the next setting (`-Wconf <filters>:<action> <filters>:<action> -nextSetting`), but it's not recommended.

The `:` before `<action>` and `&` to conjunct filters could be changed.